### ISSUE TITLE
ADC value to °C equation

### ISSUE THREAD

**Issue by msalat** [ripred/CPUTemp#3]
For the ATmega328P:

`return (wADC - 292.31 ) / 1.22;`

In the [datasheet](https://ww1.microchip.com/downloads/aemDocuments/documents/MCU08/ProductDocuments/DataSheets/ATmega48A-PA-88A-PA-168A-PA-328-P-DS-DS40002061B.pdf) chapter 24.8 is a "typical case" for 3 Temperatures in mV. AREF ist typically 1.1V .

This "typical case" approximates to temperature = `wADC/1023 * AREF * 942.485 -272.389`

Perhaps a AREF & offset parameterization should be possible.

Alongside... my two 328p's have AREF= 1.059V & 1.056V bandgap voltage and +65°C offset finally. AREF could be measured.

**Comment by msalat** [comment]
PS: with two 1n4148 diodes and 47k resistor from 3.3V, you'll get -4.7mV/°C instead of +1mV/°C — a much higher sensitivity.

**Comment by ripred** [comment]
Agreed. I changed the conversion to use the datasheet-derived voltage curve, added optional aref and offset calibration parameters, and released it as 1.0.4

https://github.com/ripred/CPUTemp/releases/tag/v1.0.4
